hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/service/ContractCallService.java

```
@@ -0,0 +1,68 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import static com.hedera.mirror.web3.evm.exception.ResponseCodeUtil.emptyFailProcessingResponse;
+import static com.hedera.mirror.web3.evm.exception.ResponseCodeUtil.getStatusOrDefault;
+
+import javax.inject.Named;
+import lombok.RequiredArgsConstructor;
+import org.apache.tuweni.bytes.Bytes;
+
+import com.hedera.mirror.web3.evm.contracts.execution.MirrorEvmTxProcessorFacade;
+import com.hedera.mirror.web3.exception.InvalidTransactionException;
+import com.hedera.mirror.web3.service.model.CallServiceParameters;
+import com.hedera.node.app.service.evm.contracts.execution.HederaEvmTransactionProcessingResult;
+
+@Named
+@RequiredArgsConstructor
+public class ContractCallService {
+    private final MirrorEvmTxProcessorFacade mirrorEvmTxProcessor;
+
+    public String processCall(final CallServiceParameters body) {
+        final var txnResult = doProcessCall(body);
+
+        if (!txnResult.isSuccessful()) {
+            throw new InvalidTransactionException(getStatusOrDefault(txnResult));
+        }
+
+        final var callResult = txnResult.getOutput() != null
+                ? txnResult.getOutput() : Bytes.EMPTY;
+
+        return callResult.toHexString();
+    }
+
+    private HederaEvmTransactionProcessingResult doProcessCall(final CallServiceParameters body) {
+        try {
+            return
+                    mirrorEvmTxProcessor.execute(
+                            body.getSender(),
+                            body.getReceiver(),
+                            body.getProvidedGasLimit(),
+                            body.getValue(),
+                            body.getCallData(),
+                            body.isStatic());
+
+        } catch (Exception e) {
```

This swallows the exception. Shouldn't this return some response to the user or at minimum log the error?

---

Not sure whether to wrap a potential NullPointer/InvalidArgument.. exceptions returned by the TxnProcessor in `InvalidTransactionException` or to just leave them to be catched by the `ContractController`'s `genericError`?

---

The other 2 possibility  indicating: wrong input, insufficient account balance etc. are `IllegalStateException` and `IllegalArgumentException` everything else I think can be consider as a internal server error and can be be handled by:
https://github.com/hashgraph/hedera-mirror-node/blob/b786289a99fa9d38a65f370f74073035a177cb6e/hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/controller/ContractController.java#L122

---

Then we should probably catch `IllegalStateException | IllegalArgumentException` and rethrow as `InvalidTransactionException` and not catch others.